### PR TITLE
Move doc building utilities to separate file

### PR DESCRIPTION
**Pull request by AVHopp**
This PR creates a new file under docs/scripts that contains utilities for building the documentation.

The reasons for this change are mainly
1. We might need some more utilities for properly including figures in light/dark mode in the examples
2. The conversion file contained functions that are used in the conversion, but technically not part of the conversion itself (like `adjust_banners`).

The only content-wise change is that the `convert_examples` function (which now lives in `utils`) has another argument in its signature for ignoring the examples. Previously, this was not necessary as the global flag was used. Other than that, no changes were made.

### CODE REVIEW

**Review by AdrianSosic — APPROVED** [review]
Hi @AVHopp, approved. There are a few things in the PR description you should fix before merging, e.g.:
* which now lives
* ignoring the examples
* ...
Just have another look =)

**Review by RimRihana — APPROVED** [review]

**Review by Scienfitz — APPROVED** [review]

File: CHANGELOG.md
Comment (Scienfitz) [inline CHANGELOG.md:15]: not sure if a changelog entry is really required for this PR but feel free to merge

**Review by Scienfitz — APPROVED** [review]

File: CHANGELOG.md
Comment (AVHopp) [reply CHANGELOG.md:15]: Well, it is a change, hence I included this. But I do not have any hard feelings about this, if you'd say "this is to minor", I can also remove this.

Comment (AdrianSosic) [reply CHANGELOG.md:15]: Yes, but a change more like "rewrote a docstring" instead of "changed that function". Though not 100% clear in all cases, for my taste we should draw the line such that only changes are included that either 1) would be important for the user or 2) mark some architecture change that is important for us / the logic of the repo. The latter would be e.g. change from markdown to html documentation. 

But at the same time, as a general rule, having rather too much in the changelog is better than missing important parts. Just need a good compromise to avoid cluttering it...

Comment (AVHopp) [reply CHANGELOG.md:15]: Agree with your two points for how to decide whether or not to put something in the CHANGELOG. So I'd remove this again from the CHANGELOG once I have the blessing of @Scienfitz

Comment (Scienfitz) [reply CHANGELOG.md:15]: *puts blessing*

Comment (AVHopp) [reply CHANGELOG.md]: I have removed it. Will thus merge this once the tests are done.

**Pull request merged**
